sqlmesh/core/plan/evaluator.py

```diff
@@ -358,9 +359,27 @@ def _restate(self, plan: EvaluatablePlan, snapshots_by_name: t.Dict[str, Snapsho
         if not plan.restatements:
             return
 
+        snapshots_to_restate = {
+            snapshots_by_name[name].table_info for name in plan.restatements.keys()
```

Comment:
```suggestion
            snapshots_by_name[name].table_info for name in plan.restatements
```

Reply:
This gets me every time :) thanks for flagging